concent_api/core/message_handlers.py

```diff
@@ -922,6 +923,9 @@ def handle_messages_from_database(client_public_key: bytes) -> Union[message.Mes
 
         assert pending_response.response_type_enum in set(PendingResponse.ResponseType)
 
+        if not are_protocol_versions_in_related_messages_compatible(pending_response.subtask, client_public_key):
```

Comment:
@dybi decorator checks if `protocol_version` in request is compatible with concent version (missing is also compatible). If client changes his `golem_messages` version during one use case (concerning one subtask) decorator will not see it. This validation will- it compares protocol version in concent (also version in request) with previous messages in database.

@rwrzesien 
Ad1. Right, `clean` ensures that. All stored messages have the same version. It is enough to check one stored_message, I chosen `task_to_compute` as most basic message always stored.
Ad2. Right, `ForcePaymentCommitted` is only one type that don't stores message. It is checked for all other messages.

Reply:
Ok, now I understand this change, but then I think that in each case the exact message that is returned should be checked. It might still happen somehow that there will be two different versions for single subtask, and checking different message then returned one is not giving 100% sure. But this will be ugly change to put this check in every `if`. Maybe this can somehow be resolved on the queryset level?

Please fix styling around this change.

Reply:
Two different protocol versions in different stored messages for single subtask is not possible. There is validation in `clean` in Subtask checking if **all** messages have the same version. If not there is validation error and such Subtask will not be saved. I think it is save.. I don't see possibility to store subtask with messages in different versions. So check of one message is enough. And it is checked now in one place at the beginning- for all responses that return stored message it is checked `task_to_compute_version`